### PR TITLE
Example southpark distributed

### PR DESCRIPTION
**Pull request by maximilianwerk**
This adds the ability to test a distributed southpark version. It would be nice if someone can test it via:
```
cd southpark-test
./tests/distributed/test_integration.sh
```
You may need to install docker-compose beforehand (see https://docs.docker.com/compose/install/ )

### CODE REVIEW

File: southpark-search/tests/distributed/test_integration.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    -F "uses_files=@pods/index.yml" \
+    -F "pymodules_files=@pods/text_loader.py" \
+    -F "yamlspec=@tests/distributed/flow-query.yml"\
+    | jq -r .flow_id)
+
+echo "Successfully started the flow: ${FLOW_ID}"
+
+curl --request POST -d '{"top_k": 10, "mode": "search",  "data": ["text:hey, dude"]}' -H 'Content-Type: application/json' '0.0.0.0:45678/api/search'
+
+curl --request GET "http://0.0.0.0:8000/v1/flow/${FLOW_ID}" -H "accept: application/json"
```
Comment (JoanFM) [inline southpark-search/tests/distributed/test_integration.sh:29]: what does this do?

**Comment by theUnkownName** [comment]
Do I need some special access? it gives me this error: `ERROR: pull access denied for south, repository does not exist or may require 'docker login': denied: requested access to the resource is denied`

File: southpark-search/tests/distributed/.env [hunk 2]
```diff
@@ -0,0 +1,3 @@
+JINA_ENCODER_HOST=encoder
```
Comment (JoanFM) [inline southpark-search/tests/distributed/.env:1]: `encoder` is a valid `host`?

**Review by JoanFM — COMMENTED** [review]
What is the structure `approximated` in this example? It has the `flow` in one "server" and all the pods in another one?

File: southpark-search/tests/distributed/.env (hunk 2 above)
Comment (maximilianwerk) [reply southpark-search/tests/distributed/.env:1]: in docker-compose it is.

> Do I need some special access? it gives me this error: `ERROR: pull access denied for south, repository does not exist or may require 'docker login': denied: requested access to the resource is denied`

fixed.

**Comment by maximilianwerk** [comment]
> Do I need some special access? it gives me this error: `ERROR: pull access denied for south, repository does not exist or may require 'docker login': denied: requested access to the resource is denied`

Thanks for bringing this up. Fixed.

File: southpark-search/tests/distributed/test_integration.sh (hunk 1 above)
Comment (maximilianwerk) [reply southpark-search/tests/distributed/test_integration.sh:29]: Just a status check. Since we can...

**Comment by maximilianwerk** [comment]
> What is the structure `approximated` in this example? It has the `flow` in one "server" and all the pods in another one?

It is encoder and indexer pods on the remote machine. The reason is, that I had issues putting one of them on the local machine. I'll try to achieve this in a further step.

**Review by theUnkownName — APPROVED** [review]
It looks good to me

**Comment by JoanFM** [comment]
Would it be hard to go the extra step of adding a third "server" where the `index` could live?

**Comment by maximilianwerk** [comment]
> Would it be hard to go the extra step of adding a third "server" where the `index` could live?

Nope, super easy, I'll do.

**Comment by maximilianwerk** [comment]
Reasoning for fixing versions: That should be a blueprint for now for the terraform usage and in the jina CI. Thus, I want to avoid having non-running parts, when some updates happening to jina/jinad for now. In the future, the current jina/jinad versions should be used.

File: southpark-search/tests/distributed/test_integration.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+set -e
+
+if [ "${PWD##*/}" != "southpark-search" ]
+  then
+    echo "test_integration.sh should only be run from the southpark-search base directory"
+    exit 1
+fi
+
+export $(grep -v '^#' tests/distributed/.env | xargs -d '\n')
```
Comment (JoanFM) [inline southpark-search/tests/distributed/test_integration.sh:9]: is this the line passing the `env variables`?

Comment (maximilianwerk) [reply southpark-search/tests/distributed/test_integration.sh:10]: no, It sets them locally for this shell context. added a comment.

**Pull request merged**